internal/pkg/cli/env_delete.go

```diff
@@ -124,66 +148,89 @@ func (opts *DeleteEnvOpts) Validate() error {
 	}
+	var names []string
+	for _, env := range envs {
+		names = append(names, env.Name)
+	}
+	name, err := o.prompt.SelectOne(envDeleteNamePrompt, "", names)
+	if err != nil {
+		return fmt.Errorf("prompt for environment name: %w", err)
+	}
+	o.EnvName = name
 	return nil
 }
 
-// RecommendedActions is a no-op for this command.
-func (opts *DeleteEnvOpts) RecommendedActions() []string {
+func (o *DeleteEnvOpts) askProfile() error {
+	if o.EnvProfile != "" {
+		return nil
+	}
+
+	profile, err := o.prompt.Get(
+		fmt.Sprintf(fmtEnvDeleteProfilePrompt, color.HighlightUserInput(o.EnvName)),
+		envDeleteProfileHelpPrompt,
+		nil, // no validation needed
+		prompt.WithDefaultInput("default"))
+	if err != nil {
+		return fmt.Errorf("prompt to get the profile name: %w", err)
+	}
+	o.EnvProfile = profile
 	return nil
 }
 
-func (opts *DeleteEnvOpts) shouldDelete(projName, envName string) (bool, error) {
-	if opts.SkipConfirmation {
+func (o *DeleteEnvOpts) shouldDelete(projName, envName string) (bool, error) {
+	if o.SkipConfirmation {
 		return true, nil
 	}
 
-	shouldDelete, err := opts.prompt.Confirm(fmt.Sprintf(fmtDeleteEnvPrompt, envName, projName), "")
+	shouldDelete, err := o.prompt.Confirm(fmt.Sprintf(fmtDeleteEnvPrompt, envName, projName), "")
 	if err != nil {
 		return false, fmt.Errorf("prompt for environment deletion: %w", err)
 	}
 	return shouldDelete, nil
 }
 
 // deleteStack returns true if the stack was deleted successfully. Otherwise, returns false.
-func (opts *DeleteEnvOpts) deleteStack() bool {
-	opts.prog.Start(fmt.Sprintf(fmtDeleteEnvStart, opts.env.Name, opts.env.Project))
-	if err := opts.deployClient.DeleteEnvironment(opts.env.Project, opts.env.Name); err != nil {
-		opts.prog.Stop(fmt.Sprintf(fmtDeleteEnvFailed, opts.env.Name, opts.env.Project, err))
+func (o *DeleteEnvOpts) deleteStack() bool {
+	o.prog.Start(fmt.Sprintf(fmtDeleteEnvStart, o.EnvName, o.ProjectName()))
+	if err := o.deployClient.DeleteEnvironment(o.ProjectName(), o.EnvName); err != nil {
+		o.prog.Stop(fmt.Sprintf(fmtDeleteEnvFailed, o.EnvName, o.ProjectName(), err))
 		return false
 	}
-	opts.prog.Stop(fmt.Sprintf(fmtDeleteEnvComplete, opts.env.Name, opts.env.Project))
+	o.prog.Stop(fmt.Sprintf(fmtDeleteEnvComplete, o.EnvName, o.ProjectName()))
 	return true
 }
 
-func (opts *DeleteEnvOpts) deleteFromStore() {
-	if err := opts.storeClient.DeleteEnvironment(opts.env.Project, opts.env.Name); err != nil {
-		log.Infof("Failed to remove environment %s from project %s store: %w\n", opts.env.Name, opts.env.Project, err)
+func (o *DeleteEnvOpts) deleteFromStore() {
+	if err := o.storeClient.DeleteEnvironment(o.ProjectName(), o.EnvName); err != nil {
+		log.Infof("Failed to remove environment %s from project %s store: %w\n", o.EnvName, o.ProjectName(), err)
 	}
 }
 
+// initDeleteEnvClientsFromProfile is overriden in tests.
+var initDeleteEnvClientsFromProfile = func(o *DeleteEnvOpts) error {
```

Comment:
I think we should plan to replace this with your suggestion of having mockable providers for clients so that we can assert we're making the proper calls instead of no-op'ing this in tests.

Reply:
Ok, I think I got a bit creative here. Let me know what you think!

Instead of having a global variable, I moved the function to be a field within the struct. This keeps the package scope clean and moves the jankiness to the struct. It's not great but I think it's better than before.